DataConnectors/Agari/AzureFunctionAgari/run.ps1

```diff
@@ -0,0 +1,362 @@
+        }
+        # Get OAuth 2.0 Token
+        $tokenRequest = Invoke-WebRequest -Method Post -Uri $uri -ContentType "application/x-www-form-urlencoded" -Body $body -UseBasicParsing
+        # Unpack Access Token
+        $sgapi_token = ($tokenRequest.Content | ConvertFrom-Json).access_token
+        # Base URL
+        $sgapi_uri = 'https://graph.microsoft.com/beta/security/tiIndicators'
+        $sgapi_headers = @{
+                            'Authorization' = "Bearer $sgapi_token"
+                            'ContentType' = 'application/json'
+                        }
+}
+
+#Set uris for get/revoke bearer tokens
+$get_apd_token_uri='https://api.agari.com/v1/ep/token'
+$get_bp_token_uri='https://api.agari.com:443/v1/cp/oauth/token'
+$get_apr_token_uri='https://api.agari.com/v1/apr/token'
+$revoke_apd_token_uri='https://api.agari.com/v1/ep/revoke'
+$revoke_apr_token_uri='https://api.agari.com/v1/apr/revoke'
+$revoke_bp_token_uri='https://api.agari.com/v1/cp/oauth/revoke'
+
+
+# Get the Bearer Tokens
+if ($bpEnabled){
+    #Get the bearer token
+    $bpToken = GetToken $client_id $client_secret $get_bp_token_uri
+    # set the headers
+    if ($bpToken){
+        $bpHeaders = @{
+                    'Authorization'="Bearer $bpToken"
+                    'ContentType' = 'application/json'
+                }
+    } else { 
+        $bpToken = $false
+        Write-Host "Error Fetching BP Token" 
+        }
+} 
+if ($adEnabled){
+    #Get the bearer token
+    $apdToken = GetToken $client_id $client_secret $get_apd_token_uri
+    # set the headers
+    if ($apdToken){
+        $apdHeaders = @{
+                    'Authorization'="Bearer $apdToken"
+                    'ContentType' = 'application/json'
+                }
+    } else { 
+        $apdToken = $false
+        Write-Host "Error Fetching APD Token" 
+        }
+}
+if ($aprEnabled){
+    #Get the bearer token
+    $aprToken = GetToken $client_id $client_secret $get_apr_token_uri
+    # set the headers
+    if ($aprToken){
+        $aprHeaders = @{
+                    'Authorization'="Bearer $aprToken"
+                    'ContentType' = 'application/json'
+                }
+    } else { 
+        $aprToken = $false
+        Write-Host "Error Fetching APR Token" 
+        }
+}
+#Set global api variables
+$startdate=(Get-Date (get-date).addMinutes(-6) -UFormat "+%Y-%m-%dT%H:%M:01.00Z")
+$enddate=(Get-Date (get-date).addMinutes(-1) -UFormat "+%Y-%m-%dT%H:%M:00.00Z")
+$limit = 200
+$offset = 0
+$TimeStampField = "DateValue"
+
+# --------------------------------------------------- #
+# Below is for loading BP Logs and TIs into Sentinel. #
+# --------------------------------------------------- #
+
+#Check if BP is enabled and we got a token
+if (($bpEnabled) -and ($bpToken)) {
+    $offset = 0
+    do {
+        $BPAlertsListAPI = "https://api.agari.com/v1/cp/alert_events?start_date=$startdate&end_date=$enddate&fields=id%2C&limit=$limit&offset=$offset&sort=created_at%20ASC"
```

Comment:
Nice use of &fields here to reduce payload size

Reply:
Thanks!  If you got the user-agent string I can add it to the web-requests.

Reply:
Doh, forgot that.
"AgariSentinel BP/APD_Integration/v1.0 MicrosoftSentinel/v?.?"

Any chance you can get the hosting Sentinel version programmatically?